File: go/o_auth_app.go
// this file is @generated
package meteroid

import "time"

// An OAuth application registered by a platform
type OAuthApp struct {
	ClientId string `json:"client_id"`

	ClientSecretHint string `json:"client_secret_hint"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Id OAuthAppId `json:"id"`

	IsActive bool `json:"is_active"`

	Name string `json:"name"`

	OrganizationId OrganizationId `json:"organization_id"`

	RedirectUris RequiredSlice[string] `json:"redirect_uris"`

	Scopes RequiredSlice[string] `json:"scopes"`

	// RFC 3339 timestamp.
	UpdatedAt *time.Time `json:"updated_at,omitempty"`
}

File: python/meteroid/models/o_auth_app.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .o_auth_app_id import OAuthAppId
from .organization_id import OrganizationId


@dataclasses.dataclass
class OAuthApp(BaseModel):
    """An OAuth application registered by a platform"""

    client_id: str

    client_secret_hint: str

    created_at: datetime

    id: OAuthAppId

    is_active: bool

    name: str

    organization_id: OrganizationId

    redirect_uris: t.List[str]

    scopes: t.List[str]

    updated_at: t.Optional[datetime] = None
